Updates to include count & density to fix the use of geom_hexbin

mikebirdgeneau opened the pull request:
Updates to include count & density to fix the use of geom_hexbin with ..density.. or ..count.. 

Fixes #1608.

hadley commented on R/hexbin.R:
```diff
@@ -31,6 +31,8 @@ hexBinSummarise <- function(x, y, z, binwidth, fun = mean, fun.args = list(), dr
 
   # Convert to data frame
   out <- as.data.frame(hexbin::hcell2xy(hb))
+  out$count = as.vector(hb@count)
```
Can you please use `<-` instead of `=`?

hadley replied:
Also I'm not sure that the code should go here - it's only needed for hexbin, not the other summary functions.

hadley commented:
Can you please also add:

* A bullet point to news
* A unit test. Use the simplest possible dataset and get the computed data with `layer_data()`

mikebirdgeneau commented:
I've made the requested changes to the best of my ability (still new to pull requests and contributions to third party repos). 

- Replaced `=` with `<-`.
- Added bullet in 'news'.
- Added simple unit test for `geom_hex()`
- Noticed that the `value` property retains the same output as `count`, so really only `density` was missing as far as I can tell.

As for your comment regarding if this is the right place for the code, that was why I didn't submit a pull request in the first place on #1608 - I thought there might have been an architectural reason for the change that I wasn't familiar enough with to make the fix.

Happy to help if this is useful, otherwise I'm equally happy if someone else has a better or more elegant way to resolve this. Cheers!

hadley commented:
Can you move the code to `StatBinhex$compute_group()`?  You'll need to save the results of `hexBinSummarise()` to a variable, and then modify that.

Otherwise looks great!

mikebirdgeneau commented:
Done. I've moved the code to `StatBinhex$compute_group()`. Thanks for pointing me in the right direction @hadley. If there's anything else on this, just let me know.

------
To confirm this did indeed resolve #1608 (in addition to the successful unit test), I ran the code in #1608 and here's the result, which looks good.
```r
DF <- data.frame(x=rnorm(1000), y=rnorm(1000))
ggplot(DF) + geom_hex(aes(x=x, y=y, fill=..density..))
```
![image](https://cloud.githubusercontent.com/assets/4763142/17371941/1c891c52-595f-11e6-983c-76a1812be69f.png)

hadley commented on NEWS:
```diff
@@ -7,6 +7,9 @@ ggplot2 1.0.1
 * Improvements to order of colours and legend display for continuous colour
   scales extracted from colorbrewer palettes by `scale_*_distiller()` (@jiho, 1076)
 
+* Restore functionality for use of `..density..` in 
```
This should go at the top of `NEWS.md` (just underneath the version number heading)

hadley commented:
Nearly there - thanks for all the hard work!

hadley commented on NEWS:
```diff
@@ -2,6 +2,9 @@
 ggplot2 1.0.1
 ----------------------------------------------------------------
 
+* Restore functionality for use of `..density..` in 
```
Still the wrong file - needs to be in `NEWS.md`

hadley commented:
Thanks!

mikebirdgeneau commented:
No problem - thanks for your patience! (Appreciate you taking the time to 'teach' when it could be faster to simply resolve yourself.)
